logger/logger.go

```diff
@@ -83,36 +87,89 @@ func (l *Logger) SetServiceName(name string) {
 	l.ServiceName = name
 }
 
-func (l *Logger) Info(params *LoggerEventParams) *LoggerEvent {
-	return l.getLoggerEvent("info", params)
+func (l *Logger) Context(id LoggerEventID, extra ...ExtraData) *LoggerEvent {
+	var extraData *ExtraData
+
+	if len(extra) > 0 {
+		extraData = &extra[0]
+	}
+
+	return &LoggerEvent{
+		logLevel: "info",
+		logger:   l,
+		id:       id,
+		extra:    extraData,
+	}
+}
+
+func (l *Logger) Info() *LoggerEvent {
+	return l.getLoggerEvent("info")
 }
 
-func (l *Logger) Debug(params *LoggerEventParams) *LoggerEvent {
-	return l.getLoggerEvent("debug", params)
+func (l *Logger) Debug() *LoggerEvent {
+	return l.getLoggerEvent("debug")
 }
 
-func (l *Logger) Warn(params *LoggerEventParams) *LoggerEvent {
-	return l.getLoggerEvent("warn", params)
+func (l *Logger) Warn() *LoggerEvent {
+	return l.getLoggerEvent("warn")
 }
 
-func (l *Logger) Error(params *LoggerEventParams) *LoggerEvent {
-	return l.getLoggerEvent("error", params)
+func (l *Logger) Error() *LoggerEvent {
+	return l.getLoggerEvent("error")
+}
+
+func (event *LoggerEvent) Info() *LoggerEvent {
+	params := []interface{}{event.id, event.extra}
+	return event.logger.getLoggerEvent("info", params...)
+}
+
+func (event *LoggerEvent) Debug() *LoggerEvent {
+	params := []interface{}{event.id, event.extra}
+	return event.logger.getLoggerEvent("debug", params...)
+}
+
+func (event *LoggerEvent) Warn() *LoggerEvent {
+	params := []interface{}{event.id, event.extra}
+	return event.logger.getLoggerEvent("warn", params...)
+}
+
+func (event *LoggerEvent) Error() *LoggerEvent {
+	params := []interface{}{event.id, event.extra}
+	return event.logger.getLoggerEvent("error", params...)
+}
+
+func (event *LoggerEvent) Extra(data ...interface{}) *LoggerEvent {
+	if event.extra == nil {
+		newExtra := make(ExtraData, len(data)/2)
+		event.extra = &newExtra
+	}
+
+	extra := make(ExtraData, len(data)/2)
```

Comment:
I think it should be written like that, if new Extra should be merged with an old one:
```go
if event.extra == nil {
    newExtra := make(ExtraData, len(data)/2)
    event.extra = &newExtra
}

for i := 0; i < len(data); i += 2 {
    event.extra[data[i].(string)] = data[i+1]
}

return event
```

Or like this, if all old data should be overwritten:
```go
newExtra := make(ExtraData, len(data)/2)
event.extra = &newExtra

for i := 0; i < len(data); i += 2 {
    event.extra[data[i].(string)] = data[i+1]
}

return event
```